### PR TITLE
[RPA-912] Add optional worksheet field to Excel action Get cell

### PR DESCRIPTION
**Pull request by star-in-the-jar**
## Description
image
if worksheet is given the session is switched, value is checked and session gets switched back

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

**Review by star-in-the-jar — COMMENTED** [review]
self reviewed

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts [hunk 1]
```diff
@@ -77,7 +78,12 @@ export default class ExcelActionHandler extends StatefulActionHandler {
     async getCell(
         input: ExcelGetCellActionInput
     ): Promise<string | number> {
-        return this.session.ActiveSheet.Range(`${input.column}${input.row}`).Value();
+        const optionalWorksheet = input?.worksheet;
+        const openedWorksheet = this.session.ActiveSheet.name;
+        if (optionalWorksheet) this.session.Worksheets(optionalWorksheet).Activate();
+        const cellValue = this.session.ActiveSheet.Range(`${input.column}${input.row}`).Value();
+        if (optionalWorksheet) this.session.Worksheets(openedWorksheet).Activate();
+        return cellValue;
```
Comment (Klau-Sam) [inline runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: can't it be simplified like: this.session.Worksheets(optionalWorksheet).Range(`${input.column}${input.row}`).Value(), if optional worksheet exists?

**Review by Klau-Sam — APPROVED** [review]

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: good point

Comment (star-in-the-jar) [reply runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: it does work!
done

**Review by star-in-the-jar — COMMENTED** [review]
self review

**Pull request merged**